xds/internal/client/client.go

```diff
@@ -129,14 +130,61 @@ type loadReportingOptions struct {
 // resource updates from an APIClient for a specific version.
 type UpdateHandler interface {
 	// NewListeners handles updates to xDS listener resources.
-	NewListeners(map[string]ListenerUpdate)
+	NewListeners(map[string]ListenerUpdate, UpdateMetadata)
 	// NewRouteConfigs handles updates to xDS RouteConfiguration resources.
-	NewRouteConfigs(map[string]RouteConfigUpdate)
+	NewRouteConfigs(map[string]RouteConfigUpdate, UpdateMetadata)
 	// NewClusters handles updates to xDS Cluster resources.
-	NewClusters(map[string]ClusterUpdate)
+	NewClusters(map[string]ClusterUpdate, UpdateMetadata)
 	// NewEndpoints handles updates to xDS ClusterLoadAssignment (or tersely
 	// referred to as Endpoints) resources.
-	NewEndpoints(map[string]EndpointsUpdate)
+	NewEndpoints(map[string]EndpointsUpdate, UpdateMetadata)
+}
+
+// ServiceStatus is the status of the update.
+type ServiceStatus int
+
+const (
+	// ServiceStatusUnknown is the default state, before a watch is started for
+	// the resource.
+	ServiceStatusUnknown ServiceStatus = iota
```

Comment:
Would we ever use this value? Because when the watch is registered, we are going to set the state to `Requested`.

Reply:
Right, in practice, it will never be used.

It's defined for completeness. There's a state `unknown` defined in the proto as well, and CSDS service handler will turn this into that.